scvi/external/scbasset/_module.py

```diff
@@ -0,0 +1,334 @@
+from typing import Callable, Dict, NamedTuple, Optional
+
+import numpy as np
+import torch
+import torchmetrics
+from torch import nn
+
+from scvi.module.base import BaseModuleClass, LossOutput, auto_move_data
+
+
+class _REGISTRY_KEYS_NT(NamedTuple):
+    X_KEY: str = "X"
+    BATCH_KEY: str = "batch"
+    DNA_CODE_KEY: str = "dna_code"
+
+
+REGISTRY_KEYS = _REGISTRY_KEYS_NT()
+
+
+def _round(x):
+    return int(np.round(x))
+
+
+class _ConvBlock(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        pool_size: int = None,
+        batch_norm: bool = True,
+        dropout: float = 0.0,
+        activation_fn: Optional[Callable] = None,
+    ):
+        super().__init__()
+        self.conv = nn.Conv1d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            padding="same",
+            bias=False,
+        )
+        self.batch_norm = nn.BatchNorm1d(out_channels) if batch_norm else nn.Identity()
```

Comment:
Looks like the default `eps` in keras is `0.001` while its `0.00001` for pytorch.

https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html

https://keras.io/api/layers/normalization_layers/batch_normalization/

Reply:
I'm going to leave all of this for another PR as I'm pretty sure all the linear layer initializations are different from the authors original code